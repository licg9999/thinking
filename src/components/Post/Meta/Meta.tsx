import React from "react";

import * as styles from "./Meta.module.scss";

interface Props {
  date: string;
}

const Meta: React.FC<Props> = ({ date }: Props) => (
  <div className={styles.meta}>
    <p className={styles.date}>
      Published{" "}
      {new Date(date).toLocaleDateString("en-GB", {
        year: "numeric",
        month: "short",
        day: "numeric",
        hour: "numeric",
        minute: "numeric",
      })}
    </p>
  </div>
);

export default Meta;
