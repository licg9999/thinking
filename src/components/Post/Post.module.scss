@import "src/assets/scss/variables";
@import "src/assets/scss/mixins";

.post {
  .content {
    @include margin-auto();
  }

  .footer {
    max-width: $layout-post-width;
    @include padding(0, 0.5);
    @include margin-auto();
  }

  .comments {
    max-width: $layout-post-width;
    @include padding(0, 0.5);
    @include margin-auto();
  }

  .topping {
    @include margin(0, 0.5);
  }

  .buttons {
    align-items: center;
    display: flex;
    justify-content: left;
    margin: #{$typographic-leading + "px"} auto 0;
    max-width: $layout-post-width;
    text-align: center;

    .buttonArticles {
      @include margin(0, 0.5, 0, 0);
    }
  }
}

@include breakpoint-md {
  .post {
    .footer {
      @include padding-equal(0);
    }

    .comments {
      @include padding-equal(0);
    }
  }
}

@include breakpoint-lg {
  .post {
    .buttons {
      left: 30px;
      position: fixed;
      top: 30px;
      @include margin(0, 0.5, 0, 0);
    }
  }
}
