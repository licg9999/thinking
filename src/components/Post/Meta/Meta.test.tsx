import React from "react";

import { Meta } from "@/components/Post/Meta";
import * as mocks from "@/mocks";
import * as testUtils from "@/utils/test-utils";

describe("Meta", () => {
  test("renders correctly", () => {
    const props = {
      date: mocks.markdownRemark.frontmatter.date,
    };

    const tree = testUtils
      .createSnapshotsRenderer(<Meta {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
