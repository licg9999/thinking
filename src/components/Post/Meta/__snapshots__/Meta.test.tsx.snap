// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meta renders correctly 1`] = `
<div>
  <p>
    Published
     
    1 Sept 2016, 01:00
  </p>
</div>
`;
